ui/src/components/admin/announcements/AdminAnnouncementContainer.js

```diff
@@ -0,0 +1,356 @@
+      setIsLoading(false);
+    } catch (err) {
+      console.log(err);
+      setIsLoading(false);
+    }
+  };
+
+  const onDelete = async (e) => {
+    try {
+      setIsLoading(true);
+      if (window.confirm('Are you sure you want to delete this?')) {
+        const { status } = await axios.put('/api/announcement/update', {
+          id: e.target.value,
+          isDeleted: true,
+        });
+        if (status === 200) {
+          toast({
+            description: 'Announcement has been deleted',
+            status: 'success',
+            duration: 8000,
+            isClosable: true,
+          });
+        }
+      }
+
+      await getAnnouncementList;
+      setIsLoading(false);
+    } catch (err) {
+      console.log(err);
+      setIsLoading(false);
+    }
+  };
+
+  const onPublish = async (e) => {
+    try {
+      setIsLoading(true);
+      const announcementId = e.target.value;
+      const announcementData = announcementList.find(
+        (item) => item.id === announcementId
+      );
+
+      const { status } = await axios.put('/api/announcement/update', {
+        id: announcementId,
+        isPublished: !announcementData.isPublished,
+      });
+
+      if (status === 200) {
+        toast({
+          description: 'Announcement has been updated',
+          status: 'success',
+          duration: 8000,
+          isClosable: true,
+        });
+      } else {
+        toast({
+          description: 'There was an issue with the request',
+          status: 'error',
+          duration: 8000,
+          isClosable: true,
+        });
+      }
+
+      await getAnnouncementList();
+      setIsLoading(false);
+    } catch (err) {
+      console.log(err);
+      setIsLoading(false);
+    }
+  };
+
+  const showProperDate = (startDate, endDate) => {
+    if (startDate && endDate) {
+      return `${startDate} - ${endDate}`;
+    }
+    if (!startDate && !endDate) {
+      return '-';
+    }
+    return startDate;
+  };
+
+  const showProperTime = (startTime, endTime) => {
+    if (startTime && endTime) {
+      return `${startTime} - ${endTime}`;
+    }
+    if (!startTime && !endTime) {
+      return '-';
+    }
+    return startTime;
+  };
+
+  return (
+    <Container maxW="container.xl">
+      <Heading as="h1" size="xl" pb={3}>
+        Announcements Manager
+      </Heading>
+      <Stack direction="row">
+        <Button colorScheme="blue" size="lg" onClick={onCreate}>
+          Add New
+        </Button>
+        <Button colorScheme="blue" size="lg" onClick={toggleAnnouncementsView}>
+          {isCurrentAnnouncements
+            ? 'Past Announcements'
+            : 'Current Announcements'}
+        </Button>
+      </Stack>
+      <Heading as="h2" size="lg" pt={7}>
+        {isCurrentAnnouncements
+          ? 'Current Announcements'
+          : 'Past Announcements'}
+      </Heading>
+      <List spacing="2" pt={3}>
+        {/* List announcements */}
+        {announcementList.map((announcementItem) => (
```

Comment:
woudl it be better to use agGrid to show these announcements? feel like it's easier to edit that way, and format will be much more similar to what we get from google docs

Reply:
agGrid not as user friendly cuz we may extend it to non-tech people for delegation - Andrea, CXO